frankenstein/mock_dynamodb.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package frankenstein
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+type mockDynamoDB struct {
+	mtx    sync.RWMutex
+	tables map[string]mockDynamoDBTable
+}
+
+type mockDynamoDBTable struct {
+	hashKey  string
+	rangeKey string
+	items    map[string][]mockDynamoDBItem
+}
+
+type mockDynamoDBItem map[string]*dynamodb.AttributeValue
+
+func newMockDynamoDB() *mockDynamoDB {
+	return &mockDynamoDB{
+		tables: map[string]mockDynamoDBTable{},
+	}
+}
+
+func (m *mockDynamoDB) CreateTable(input *dynamodb.CreateTableInput) (*dynamodb.CreateTableOutput, error) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	var hashKey, rangeKey string
+	for _, schemaElement := range input.KeySchema {
+		if *schemaElement.KeyType == "HASH" {
+			hashKey = *schemaElement.AttributeName
+		} else if *schemaElement.KeyType == "RANGE" {
+			rangeKey = *schemaElement.AttributeName
+		}
+	}
+
+	m.tables[*input.TableName] = mockDynamoDBTable{
+		hashKey:  hashKey,
+		rangeKey: rangeKey,
+		items:    map[string][]mockDynamoDBItem{},
+	}
+
+	return &dynamodb.CreateTableOutput{}, nil
+}
+
+func (m *mockDynamoDB) ListTables(*dynamodb.ListTablesInput) (*dynamodb.ListTablesOutput, error) {
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+
+	var tableNames []*string
+	for tableName, _ := range m.tables {
+		func(tableName string) {
+			tableNames = append(tableNames, &tableName)
+		}(tableName)
+	}
+	return &dynamodb.ListTablesOutput{
+		TableNames: tableNames,
+	}, nil
+}
+
+func (m *mockDynamoDB) BatchWriteItem(input *dynamodb.BatchWriteItemInput) (*dynamodb.BatchWriteItemOutput, error) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	for tableName, writeRequests := range input.RequestItems {
+		table, ok := m.tables[tableName]
+		if !ok {
+			return &dynamodb.BatchWriteItemOutput{}, fmt.Errorf("table not found")
+		}
+
+		for _, writeRequest := range writeRequests {
+			hashValue := *writeRequest.PutRequest.Item[table.hashKey].S
+			rangeValue := writeRequest.PutRequest.Item[table.rangeKey].B
+			fmt.Printf("Write %s/%x\n", hashValue, rangeValue)
```

Comment:
Is this still supposed to be here? (same below)

I guess it's helpful debug output when a test fails, but maybe then use the normal Prometheus logging lib?